builtins/amp-ad.js

```diff
@@ -263,6 +263,64 @@ export function installAd(win) {
         this.element.removeChild(this.iframe_);
       });
     }
+
+    /**
+     * Takes an AMP ad and immediately renders.
+     * @param {string} ampHtmlAd AMP HTML that will be immediately rendered if
+     *   it is validated.
+     * @private
+     */
+    validateAndRenderOrFallback_(ampHtmlAd) {
+      // First find the body of the valid AMP document. Wrap this in an
+      // identifiable div and append it as a child to our amp-ad tag.
+      var ampBodyStart = ampHtmlAd.indexOf("<body>");
```

Comment:
Also verify the existence of </body>?  Other option is to use regexp capture, ie the following /<body>(.*)<\/body>/.exec('<body>foo</body>')[1] returns 'foo'